File: api/Pocketree.Api/Models/Entities/Task.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Globalization;

namespace ADproject.Models.Entities
{
    public class Task
    {
        [Key]
        public int TaskID { get; set; }
        [Required]
        public string? Description { get; set; }
        [Required]
        public string? Difficulty { get; set; } = "Easy";
        [Required]
        public int CoinReward { get; set; } = 10;
        [Required]
        public bool RequiresEvidence { get; set; } = false; // For ML Verification
        [Required, StringLength(255)]
        public string? Keyword { get; set; } = ""; // For ML use
        [Required, StringLength(50)]
        public string? Category { get; set; } = "General"; // For ML use
        [Required, StringLength(255)]
        public string? NegativeKeyword { get; set; } = ""; // For ML use
        [Required]
        public string? SourceType { get; set; } = "Normal"; // For ML use

        [NotMapped]
        public bool isCompleted { get; set; } // Not created in DB, field is just to match Android's side
        [NotMapped]
        public bool isPassed {get; set;}
    }
}

File: android/app/src/main/java/com/pocketree/app/Task.kt
package com.pocketree.app

import com.google.gson.annotations.SerializedName

data class Task (
    @SerializedName("taskID") val taskID: Int,
    @SerializedName("description") val description: String,
    @SerializedName("isCompleted") var isCompleted:Boolean = false,
    @SerializedName("isPassed") var isPassed:Boolean = false,
    @SerializedName("difficulty") val difficulty: String,
    @SerializedName("coinReward") var coinReward: Int,
    @SerializedName("requiresEvidence") val requiresEvidence: Boolean,
    @SerializedName("keyword") val keyword: String?, //for ML
    @SerializedName("category") val category: String? // for ML
    )